conda_build/index.py

```diff
@@ -1179,20 +1179,24 @@ def index_subdir(self, subdir, index_file=None, verbose=False, progress=False):
                     json.dump(stat_cache, fh)
         return new_repodata
 
-    def _ensure_dirs(self, subdir):
+    def _ensure_dirs(self, subdir: str):
+        """Create cache directories within a subdir.
+
+        Args:
+            subdir (str): name of the subdirectory
+        """
         # Create all cache directories in the subdir.
-        ensure = lambda path: isdir(path) or os.makedirs(path)
-        cache_path = join(self.channel_root, subdir, ".cache")
-        ensure(cache_path)
-        ensure(join(cache_path, "index"))
-        ensure(join(cache_path, "about"))
-        ensure(join(cache_path, "paths"))
-        ensure(join(cache_path, "recipe"))
-        ensure(join(cache_path, "run_exports"))
-        ensure(join(cache_path, "post_install"))
-        ensure(join(cache_path, "icon"))
-        ensure(join(self.channel_root, "icons"))
-        ensure(join(cache_path, "recipe_log"))
+        cache_path = Path(self.channel_root, subdir, ".cache")
+        cache_path.mkdir(exist_ok=True)
```

Comment:
`os.makedirs(path)` == `Path(path).mkdir(parents=True)`

```suggestion
        cache_path.mkdir(parents=True, exist_ok=True)
```